flask/mongo.py

```diff
@@ -1,16 +1,55 @@
 import os
+from datetime import datetime, timedelta
 from pymongo import MongoClient
 from dotenv import load_dotenv
 from gridfs import GridFS
 from app import app
+from flask import send_file
+import subprocess
 
 load_dotenv()
 database_name = os.environ['MONGO_DB_NAME']
 try:
     mongo_uri = os.environ['MONGO_URI']
+    mongo_username = os.environ['MONGO_USER_NAME']
+    mongo_password = os.environ['MONGO_PASSWORD']
     client = MongoClient(mongo_uri)
 except KeyError as err:
     print("Add MONGO_URI to .env file")
 
 db = client[database_name]
 file_storage = GridFS(db, collection="tarballs")
+
+
+@app.route("/registry/clone", methods=["GET"])
```

Comment:
Can you test if this API crashes in vercel deployment ?
Also, should be consider making this API authorized ?

Reply:
yes, as vercel has headless running thus as we are storing the archive in a static folder , I have tested It runs well in the docker containers. I think according to the general requirements of the https://repology.org/docs/requirements , we should make this open to everyone.